de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/EditorManager.java

```diff
@@ -449,7 +453,36 @@ private void endSession() {
             openEditorPaths.clear();
         }
 
+        private void notifyUser(
```

Comment:
Misplaced ?  The EditorManager is not even part of the ui package. Please find a better place.

Reply:
I agree that it seems misplaced here. To be honest, I just put it here because this deals with the main logic ending the session.

I don't know where else to put it. Maybe into the [`UserStatusChangeHandler`](https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/core/ui/eventhandler/UserStatusChangeHandler.java), but this does not seem very fitting either.

I could move it to its own class (that just contains a `ISessionLifecycleListener`), but this still leaves the problem where to put that class.
I am not sure whether the `dpp.intellij.ui` package is the right location. The package mainly contains the implementation of different UI elements. This method just uses parts of the UI to notify the user, but is not part of the UI itself. I would associate it more with the general session logic.

How about `dpp.core.ui.eventhandler` (the core package contained in the IntelliJ project, not the core project)? This would be the best place I could find while looking through the project.

(Or `dpp.core.awareness`, but this does not fit that well either.)